distribution/lib/Standard/Base/0.2.32-SNAPSHOT/src/Data/Vector.enso

```diff
@@ -747,13 +745,12 @@ type Vector
     distinct : (Any -> Any) -> Vector Any
     distinct (on = x->x) =
         builder = here.new_builder
-        existing_elements = HashSet.new
-
-        this.each elem->
-            proj = on elem
-            if existing_elements.contains proj then Nothing else
-                existing_elements.add proj
-                builder.append elem
+        this.fold Map.empty state->
```

Comment:
```suggestion
        this.fold Map.empty existing_elements->
```

As discussed, the name `state` is not very descriptive, so it would be cool to make it a bit more.